video/out/drm_atomic.c

```diff
@@ -452,3 +453,140 @@ bool drm_mode_destroy_blob(int fd, struct drm_mode *mode)
+        p->data.hdmi_metadata_type1.white_point.y = clamp16(roundf(colors.white.y * 50000.0));
+
+        // Min Luminance: 16-bit value in units of 0.0001 cd/m2, where 0x0001 represents 0.0001 cd/m2 and 0xFFFF represents 6.5535 cd/m2.
+        p->data.hdmi_metadata_type1.min_display_mastering_luminance = clamp16(floorf(min_luminance * 10000.0));
+
+        // Max Luminance: 16-bit value in units of 1 cd/m2, where 0x0001 represents 1 cd/m2 and 0xFFFF represents 65535 cd/m2.
+        p->data.hdmi_metadata_type1.max_display_mastering_luminance = clamp16(ceilf(max_luminance));
+
+        // Max Content Light Level: 16-bit value in units of 1 cd/m2, where 0x0001 represents 1 cd/m2 and 0xFFFF represents 65535 cd/m2.
+        p->data.hdmi_metadata_type1.max_cll = clamp16(ceilf(max_cll));
+
+        // Max Frame Average Light Level: 16-bit value in units of 1 cd/m2, where 0x0001 represents 1 cd/m2 and 0xFFFF represents 65535 cd/m2.
+        p->data.hdmi_metadata_type1.max_fall = clamp16(ceilf(max_fall));
+
+        /*
+        printf("HDR: peak=%f min=%d max=%d cll=%d fall=%d\n",
+            color->sig_peak,
+            p->data.hdmi_metadata_type1.min_display_mastering_luminance,
+            p->data.hdmi_metadata_type1.max_display_mastering_luminance,
+            p->data.hdmi_metadata_type1.max_cll,
+            p->data.hdmi_metadata_type1.max_fall);
+        */
+
+        drmModeCreatePropertyBlob(ctx->connector->fd, &p->data, sizeof p->data, &p->blob_id);
+        drmModeAtomicReqPtr request = drmModeAtomicAlloc();
+        if (!request)
+            return;
+        drmModeAtomicAddProperty(request, ctx->connector->id, property_hdr_id, p->blob_id);
+        drmModeAtomicCommit(ctx->connector->fd, request, DRM_MODE_ATOMIC_ALLOW_MODESET, NULL);
+        drmModeAtomicFree(request);
+    }
+
+#endif
+}
+
+// switch HDR off
+void drm_destroy_hdrmeta(struct drm_atomic_context *ctx) {
+#ifdef DRM_HAS_HDR_METADATA_INFOFRAME
+    if (ctx->hdr_metadata.blob_id) {
+        drmModeDestroyPropertyBlob(ctx->connector->fd, ctx->hdr_metadata.blob_id);
+        ctx->hdr_metadata.blob_id = 0;
+
+        drmModeAtomicReqPtr request = drmModeAtomicAlloc();
+        if (!request)
+             return;
+
+        // switch back to SDR
+        uint32_t property_hdr_id = get_property_id(ctx->connector->fd, ctx->connector->props, "HDR_OUTPUT_METADATA");
+
+        ctx->hdr_metadata.data.metadata_type = DRM_HDMI_STATIC_METADATA_TYPE1;
+        ctx->hdr_metadata.data.hdmi_metadata_type1.metadata_type = DRM_HDMI_STATIC_METADATA_TYPE1;
+        ctx->hdr_metadata.data.hdmi_metadata_type1.eotf = 0;           // SDR
+
+        drmModeCreatePropertyBlob(ctx->connector->fd, &ctx->hdr_metadata.data, sizeof ctx->hdr_metadata.data, &ctx->hdr_metadata.blob_id);
+
+        drmModeAtomicAddProperty(request, ctx->connector->id, property_hdr_id, ctx->hdr_metadata.blob_id);
+        drmModeAtomicCommit(ctx->connector->fd, request, DRM_MODE_ATOMIC_ALLOW_MODESET, NULL);
```

Comment:
Not 100% on that one, does mpv restore any other state on quit? From fuzzy memory, apps using DRM were not required ... so I'm not sure if this function is needed. Perhaps it's masking a driver bug?